addons/point_of_sale/models/pos_session.py

```diff
@@ -542,12 +545,19 @@ def _create_invoice_receivable_lines(self, data):
 
         invoice_receivable_vals = defaultdict(list)
         invoice_receivable_lines = {}
-        for receivable_account_id, amounts in invoice_receivables.items():
-            invoice_receivable_vals[receivable_account_id].append(self._get_invoice_receivable_vals(receivable_account_id, amounts['amount'], amounts['amount_converted']))
+        for key, amounts in invoice_receivables.items():
+            try:
+                receivable_account_id, payment = key
+                partner_id = payment.partner_id.id
+            except TypeError:
+                receivable_account_id = key
+                partner_id = False
```

Comment:
With consistent type you don't need the Try/except:
```suggestion
            receivable_account_id, payment = key
            partner_id = payment.partner_id.id if payment else False
```